proposals/model-management.md

```diff
@@ -0,0 +1,67 @@
+# Model Management
+
+## Problem
+
+This proposal is attempting to solve the following problems:
+
+* Slow model loading on cold-start
+* Storage of LoRA adapters generated from finetuning jobs
+* Dynamic LoRA adapter injection
+
+While working within the following constraints:
+
+* KubeAI should be able to run with zero external dependencies
+* KubeAI should be able to run on a local machine
+* KubeAI should be able to be run on clouds with heterogenous disk features (limited support for ReadOnlyMany)
+
+## Solution
+
+KubeAI should support configurable storage for both models and adapters. These settings should be configurable at the `kind: Model` level as some variants (like GCP Hyperdisk ML) are expensive and generally only recommended for larger models.
+
+```yaml
+kind: Model
+spec:
+  storageProfile:
+    model: Bucket # || None (Always Pull) || ReadWriteManyDisk (EFS/NFS) || ReadManyDisk (GCP ML Hyperdisk)
+    adapters: Bucket # || ReadWriteManyDisk (EFS/NFS)
+```
+
+Storage profiles should be defined at the KubeAI config.yaml level:
+
+```yaml
+storageProfiles:
+  Bucket:
+    bucket:
+      url: s3://...
+      # ...
+  ReadWriteManyDisk:
+    disk:
+      storageClass: aws-efs
+      # ...
+  ReadManyDisk:
+    disk:
+      storageClass: gcp-hyperdisk
```

Comment:
so would this be a single hyperdisk shared with all models? and each model is in a subdirectory?

Reply:
Updated the proposal a little more - ML Hyperdisk would be mounted once to populate it with a single model, then mounted read-only by multiple servers.